apps/studio/src/common/appdb/validators/ReadOnlyOrDefault.ts

```diff
@@ -0,0 +1,31 @@
+import { registerDecorator, ValidationArguments, ValidationOptions } from "class-validator";
+import { LicenseKey } from "../models/LicenseKey";
+import platformInfo from'@/common/platform_info';
+
+export function ReadOnlyOrDefault(validationOptions?: ValidationOptions) {
+  return function (object: Object, propertyName: string) {
+    registerDecorator({
+      name: 'readOnlyOrDefault',
+      async: true,
+      target: object.constructor,
+      propertyName: propertyName,
+      options: validationOptions,
+      validator: {
+        async validate(value: any, args: ValidationArguments) {
+          const status = await LicenseKey.getLicenseStatus();
+          const canSaveReadOnly = status.isUltimate || platformInfo.testMode;
+
+          if (canSaveReadOnly) {
+            return true;
+          }
+
+          if (value === true) {
+            (args.object as any).readOnlyMode = false;
```

Comment:
Isn't this hacky? It'd be awesome if we put this somewhere else for predictable outcome. I think TypeORM should have a way to do that? Like a setter or something. Otherwise, I guess it's fine for now.